test/lib/pgBackRestTest/Info/InfoIniUnitTest.pm

```diff
@@ -0,0 +1,467 @@
+    {
+        my $oIni = new pgBackRest::Common::Ini($strTestFile, {bLoad => false});
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $self->testResult(sub {$oIni->boolTest($strSection, $strKey)}, false, 'test bool on undefined key');
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $self->testResult(sub {$oIni->boolGet($strSection, $strKey, undef, false, false)}, false, 'get bool default false value');
+        $self->testResult(sub {$oIni->boolGet($strSection, $strKey, undef, false, true)}, true, 'get bool default true value');
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $self->testResult(sub {$oIni->boolSet($strSection, $strKey, undef, undef)}, true, 'set bool false (undef) value');
+        $self->testResult(sub {$oIni->boolGet($strSection, $strKey)}, false, '    check bool false value');
+
+        $self->testResult(sub {$oIni->boolTest($strSection, $strKey, undef, false)}, true, 'test bool on false key');
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $self->testResult(sub {$oIni->boolSet($strSection, $strKey, undef, false)}, false, 'set bool false value');
+        $self->testResult(sub {$oIni->boolGet($strSection, $strKey)}, false, '    check value');
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $self->testResult(sub {$oIni->boolSet($strSection, $strKey, undef, true)}, true, 'set bool false value');
+        $self->testResult(sub {$oIni->boolGet($strSection, $strKey)}, true, '    check value');
+
+        $self->testResult(sub {$oIni->boolTest($strSection, $strKey, undef, true)}, true, 'test bool on true key');
+    }
+
+    ################################################################################################################################
+    if ($self->begin("Ini->numericSet() & Ini->numericGet() & Ini->numericTest()"))
+    {
+        my $oIni = new pgBackRest::Common::Ini($strTestFile, {bLoad => false});
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $self->testResult(sub {$oIni->numericSet($strSection, $strKey)}, true, 'set numeric undef value');
+        $self->testResult(sub {$oIni->test($strSection, $strKey)}, false, 'test numeric undef value');
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $self->testResult(sub {$oIni->numericGet($strSection, $strKey, undef, false, 1000)}, 1000, 'get numeric default value');
```

Comment:
Just to point out again, this is the reason I am not a fan of the new set/get functions. I can't tell what the other parameters are supposed to be without digging.